src/server/decode.rs

```diff
@@ -80,6 +80,8 @@ where
     req.set_version(Some(http_types::Version::Http1_1));
 
     for header in httparse_req.headers.iter() {
+        // https://tools.ietf.org/html/rfc822#section-3.1
+        http_types::ensure_status!(header.value.is_ascii(), 400, "None ascii header");
```

Comment:
This requires two iterations over each byte — one for the is_ascii() check and the other for the from_utf8. I believe we should just map the error value for the result of from_utf8, like `std::str::from_utf8(header.value).map_err(|_| Error::from_str(400, "non-ascii header"))?`

Reply:
Sorry, I commented below already.
There is a panic that prevent mapping the error unfortunately.
Unwrap occurs here: https://github.com/http-rs/http-types/blob/41655460b0229064535308e329417f8de014c2a8/src/headers/headers.rs#L66
Error comes from here:
https://github.com/http-rs/http-types/blob/41655460b0229064535308e329417f8de014c2a8/src/headers/header_value.rs#L80